clients/drcachesim/tests/memref_gen.h

```diff
@@ -137,6 +145,41 @@ gen_exit(memref_tid_t tid)
     return memref;
 }
 
-} // namespace
+/* Returns a vector of encoded memref_t.
+ * The caller has to set tid + pid fields of the memref_t in memref_instr_t
+ * structs but not the other fields. Also note that all operand records have to
+ * be filled in for each memref when constructing memref_instr_vec.
+ */
+inline std::vector<memref_t>
+get_memrefs_from_ir(instrlist_t *ilist, std::vector<memref_instr_t> &memref_instr_vec,
+                    addr_t base_addr)
+{
+    static const int MAX_DECODE_SIZE = 1024;
+    byte decode_buf[MAX_DECODE_SIZE];
+    byte *pc =
+        instrlist_encode_to_copy(GLOBAL_DCONTEXT, ilist, decode_buf,
+                                 reinterpret_cast<app_pc>(base_addr), nullptr, true);
+    assert(sizeof(decode_buf) / sizeof(decode_buf[0]) <= MAX_DECODE_SIZE);
```

Comment:
This is checking whether the static count of elements in decode_buf is <= MAX_DECODE_SIZE? That's not very useful. I think what we need is `pc <= decode_buf + sizeof(decode_buf)`, since pc tells us how much of the buffer space we actually used.

Reply:
Done.